var/spack/repos/builtin/packages/zlib/package.py

```diff
@@ -25,27 +25,42 @@ class Zlib(Package):
             description='Produce position-independent code (for shared libs)')
     variant('shared', default=True,
             description='Enables the build of shared libraries.')
+    variant('debug',   default=False,
+            description='Build debug code (-g -O0)')
     variant('optimize', default=True,
             description='Enable -O2 for a more optimized lib')
 
+    conflicts('+debug+optimize')
+    conflicts('+shared', when='@:1.2.5')
+
     patch('w_patch.patch', when="@1.2.11%cce")
 
+    def patch(self):
+        if self.spec.satisfies('@:1.2.5'):
+            filter_file(r'stdio.h', 'stdio.h>\n#include <errno.h', 'gzio.c')
```

Comment:
Could you add a comment explaining that this is adding an `errno.h` include to `gzio.c`? This is outside what I'd consider an idiomatic usage of the `filter_file` function (a patch would be more straightforward IMO) so it took a minute to understand the intent, but a comment would be sufficient.

Reply:
It is definitely not clear to me why we wouldn't prefer a patch, especially since this change affects only a single version, so we can be absolutely sure the patch would apply correctly.